### DISCUSSION TITLE
Running your code

### DISCUSSION THREAD

**Discussion by kittaaaa** [pulsar-edit/.github#85] [General]
I want to know how to run my code on browser like VS code has live server extension i am wondering. I know Pulsar is new and I love the project and Idea and I want to help the project as a community member new to open source . Thanks

**Comment by confused-Techie** [comment]
I'm not sure how possible it'd be to run Pulsar within a Browser, what with all the native extensions being loaded, and that we are operating on a bit of an older version of Chrome at the moment. I do think we talked about it previously on the Discord but am having some trouble finding it.

It would be a fantastic feature, and if you wanted to take a shot by all means go ahead, but I think we are a little ways from that possibility as of currently.

But if anyone else from @pulsar-edit/core has anything to add please do, as I'm not as familiar with the core project as I am in other areas

**Reply by Daeraxa** [reply]
https://discord.com/channels/992103415163396136/992103415163396139/1024678625989111968 was the conversation where I brought it up before.

**Reply by confused-Techie** [reply]
For posterity, here's an excerpt from the linked conversation (in case you don't have Discord)

---

I.... don't think this will be possible in the short term, maybe not even on the long term. We have things we need to do to make this a possibility:

1. Migrate away from native modules in every possible way, or at least make then async so we can "pipe" the calls through the network
2. Migrate away from every sync call over FS (tree-view for example makes synchronous calls to the FS API)
3. Make the file watcher configurable and network transparent (that is actually quite hard to do) or allow to not use it
4. Detect when plug-ins try  to use incompatible node.js requires or modules and try to disable these plug-ins (like for example, require('fs').readFileSync and friend)

And looots of other things that I may not even know yet
For example, tree-sitter is using the Node.JS version. We would need to migrate to a web-based version; TextMate grammars suffer from a similar problem (they use Oniguruma, which is a native module that's also sync)

---

Thanks for finding the conversation @Daeraxa guess I just don't know how to use the Discord search

But that said @AyaanShaikh18 if you wanted to see Pulsar run in the browser these issues would be the place to start to do so

**Comment by mauricioszabo** [comment]
I feel that it's easier to start a new project than make Pulsar able to run on browsers.

Which, honestly, may not be _a problem_ at all - maybe when we somehow stabilize Pulsar we can somehow start a new project from scratch, copy parts of Pulsar that we feel are reusable, rewrite the ones that are not, and make it in a way that it'll be able to run in a Browser

But that is to say - it's not on short term, and probably it'll never be completed. Not because we don't want to do it, but because the structure of Pulsar is too complex and too tied on some Electron stuff that it'll be hard to make it work...

**Reply by Daeraxa** [reply]
This is a similar approach to what they are doing with Brackets from what I understand. They have Brackets as desktop app but also Phoenix which is a version specifically designed for a web browser.

**Comment by DeeDeeG** [comment]
I think this was about live previewing your HTML code that you're developing in Pulsar? And maybe VS Code has some way to prompt your browser to auto-reload/live preview HTML? (Apparently only via extensions, per VS Code's docs: https://code.visualstudio.com/Docs/languages/html#_does-vs-code-have-html-preview. As the original poster of this thread seemed to be aware.)

In fact, Atom does have some similar extensions.

https://www.sitepoint.com/7-atom-add-ons-for-running-code-and-previewing-changes/

Packages of these same names should be available in Pulsar's package API as well, since we have migrated Atom's old packages over for the time being.
